libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohortBinaryClassification.test.ts

```diff
@@ -0,0 +1,418 @@
+
+  const jointDataset = new JointDataset({
+    dataset: features,
+    localExplanations,
+    metadata: modelMetadata,
+    predictedProbabilities: probabilityY,
+    predictedY,
+    trueY
+  });
+  return jointDataset;
+}
+
+describe("Translate user defined cohorts for classification", () => {
+  const mockJointDataset = getJointDatasetBinaryClassification();
+  it("should be able to translate index cohort", () => {
+    const mockProp: IModelAssessmentDashboardProps = {
+      cohortData: [adultCohortDataIndex]
+    } as IModelAssessmentDashboardProps;
+    const [errorCohortList, errorStrings] = processPreBuiltCohort(
+      mockProp,
+      mockJointDataset
+    );
+    expect(errorStrings.length).toBe(0);
+    expect(errorCohortList.length).toBe(1);
+    expect(errorCohortList[0].cohort.name).toBe(adultCohortDataIndex.name);
+    expect(errorCohortList[0].cohort.filters.length).toBe(
+      adultCohortDataIndex.cohort_filter_list.length
+    );
+    expect(errorCohortList[0].cohort.filters[0].column).toBe(
+      adultCohortDataIndex.cohort_filter_list[0].column
+    );
+    expect(errorCohortList[0].cohort.filters[0].method).toBe(
+      adultCohortDataIndex.cohort_filter_list[0].method
+    );
+    expect(errorCohortList[0].cohort.filters[0].arg).toEqual(
+      adultCohortDataIndex.cohort_filter_list[0].arg
+    );
+  });
+  it("should be able to translate dataset continuous cohort", () => {
+    const mockProp: IModelAssessmentDashboardProps = {
+      cohortData: [adultCohortDataContinuous]
+    } as IModelAssessmentDashboardProps;
+    const [errorCohortList, errorStrings] = processPreBuiltCohort(
+      mockProp,
+      mockJointDataset
+    );
+    expect(errorStrings.length).toBe(0);
+    expect(errorCohortList.length).toBe(1);
+    expect(errorCohortList[0].cohort.name).toBe(adultCohortDataContinuous.name);
+    expect(errorCohortList[0].cohort.filters.length).toBe(
+      adultCohortDataContinuous.cohort_filter_list.length
+    );
+    for (
+      let index = 0;
+      index < errorCohortList[0].cohort.filters.length;
+      index++
+    ) {
+      expect(errorCohortList[0].cohort.filters[index].column).toContain(
+        JointDataset.DataLabelRoot
+      );
+      expect(errorCohortList[0].cohort.filters[index].method).toBe(
+        adultCohortDataContinuous.cohort_filter_list[index].method
+      );
+      expect(errorCohortList[0].cohort.filters[index].arg).toEqual(
+        adultCohortDataContinuous.cohort_filter_list[index].arg
+      );
+    }
+  });
+  it("should be able to translate dataset categorical cohort", () => {
+    const mockProp: IModelAssessmentDashboardProps = {
+      cohortData: [adultCohortDataCategorical]
+    } as IModelAssessmentDashboardProps;
+    const [errorCohortList, errorStrings] = processPreBuiltCohort(
+      mockProp,
+      mockJointDataset
+    );
+    expect(errorStrings.length).toBe(0);
+    expect(errorCohortList.length).toBe(1);
+    expect(errorCohortList[0].cohort.name).toBe(
```

Comment:
I think we are trying to make files less than 300 lines (I see a warning here), not sure if that applies to test files though, maybe this is fine